plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergResourceFactory.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.metastore.thrift.StaticMetastoreConfig;
+import io.trino.spi.TrinoException;
+import io.trino.spi.classloader.ThreadContextClassLoader;
+import io.trino.spi.connector.ConnectorSession;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.SupportsNamespaces;
+import org.apache.iceberg.hive.HiveCatalog;
+import org.apache.iceberg.io.FileIO;
+
+import javax.inject.Inject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+
+public class IcebergResourceFactory
+{
+    private final StaticMetastoreConfig metastoreConfig;
+    private final IcebergConfig icebergConfig;
+
+    @Inject
+    public IcebergResourceFactory(
+            StaticMetastoreConfig metastoreConfig,
```

Comment:
Currently I still need to set a dummy value for `hive.metastore.uri` due to the check in `StaticMetastoreConfig`, otherwise connector initialization would fail with NPE. Would be great if anyone knows how to avoid initializing that based on the config flag.